### PR TITLE
analytics for delete/restore set in 1rm tab

### PR DESCRIPTION
**Pull request by Malekai**

### CODE REVIEW

File: app/features/analysis/edit_set/OneRMEditSetActions.js [hunk 1]
```diff
@@ -75,5 +76,6 @@ const logCloseEditSetAnalytics = (state) => {
         did_change_rpe: didChangeRPE,
         did_change_tags: didChangeTags,
         did_change_reps: didChangeReps,
+        did_delete_or_restore_set: didDeleteOrRestoreSet,
```
Comment (JonnyLinja) [inline app/features/analysis/edit_set/OneRMEditSetActions.js]: - [ ] make it did_delete_set and did_restore_set

File: app/redux/selectors/AnalysisSelectors.js [hunk 2]
```diff
@@ -111,6 +113,10 @@ export const getDidUpdateMetric = (state, set) => !areEqual(set.metric, getOrigM
 
 export const getDidUpdateTags = (state, set) => !areArraysEqual(set.tags, getOrigTags(state));
 
+export const getDidDeleteSet = (state, set) => (Boolean(set.deleted) !== (getOrigDeletedFlag(state))) && !getOrigDeletedFlag(state);
+
+export const getDidRestoreSet = (state, set) => (Boolean(set.deleted) !== (getOrigDeletedFlag(state))) && getOrigDeletedFlag(state);
+
```
Comment (JonnyLinja) [inline app/redux/selectors/AnalysisSelectors.js]: - [ ] does this work if the set doesn't HAVE a deleted flag?

Comment (Malekai) [reply app/redux/selectors/AnalysisSelectors.js]: Yes, as we are checking the Boolean of set.deleted, so undefined is false.

Comment (JonnyLinja) [reply app/redux/selectors/AnalysisSelectors.js]: - [ ] However, a set is considered deleted if it's empty right? So you'd be deleting a deleted set no?

Comment (Malekai) [reply app/redux/selectors/AnalysisSelectors.js]: Updating to account for .removed

File: app/features/analysis/results/OneRMResultsActions.js [hunk 3]
```diff
@@ -21,6 +22,7 @@ export const tappedSet = (setID) => (dispatch, getState) => {
         origWeight: set.hasOwnProperty('weight') ? set.weight : null,
         origMetric: set.hasOwnProperty('metric') ? set.metric : null,
         origTags: set.hasOwnProperty('tags') ? set.tags : [],
+        origDeletedFlag: set.hasOwnProperty('deleted') ? set.deleted : SetUtils.isDeleted(set),
```
Comment (JonnyLinja) [inline app/features/analysis/results/OneRMResultsActions.js]: - [ ] doesn't SetUtils.isDeleted also check for deleted?

**Pull request merged**